### DISCUSSION TITLE
请问在WSA中有什么比较方便的启动新版lsp管理器的方法吗（lsp的活动名称是什么）

### DISCUSSION THREAD

**Discussion by KJZH001** [LSPosed/LSPosed#2729] [Q&A] [answered]
ps：我在readme中看到了不建议使用中文的提示，但是我看到有很多中文的discuss，所以我直接使用中文发送了

在老版的lsp由于会默认显示管理器的快捷方式，WSA也能正常调用它，所以没有问题

现在的版本默认只能从通知栏进入，wsa确实可以从这里进去，但是在windows上这不是一个常驻通知（点击进去一次就消失了），除非重启整个子系统，否则它都不会再次出现

![image](https://github.com/LSPosed/LSPosed/assets/29683578/9db79260-a281-467f-9b9d-ebc73b4bcdca)
正如图中所见的，lsp目前还不能兼容wsa的创建快捷方式

我记得lsp似乎也可以通过拨号键盘的暗码进入，但是wsa当然也没有拨号这种东西

目前我想到有这么两种方式：

1.（需要lsp后续更新支持）
在设置中额外添加一项，允许用户在点击通知进入后程序自动刷新一遍通知

2.通过类似adb拉起应用活动的方式启用（目前我想做的）

WsaClient.exe /launch wsa://com.android.settings/com.android.settings.Settings

WSA可以很方便的通过指定包名和活动名称的方式启动程序，我想，只要能够知道lsp管理器的对应活动名称就能直接启动它

这也是我目前想要知道的东西

感谢各位的帮助！

2024.1.4：安装Manager.apk目前看起来是我感觉最可行的方法，当然其他人也确实给出了可行的方案，但是好像目前还没有转换为wsa://这种URI的方法，所以，目前我将安装Manager.apk选定为本问题的答案

**Comment by ManyullynSword** [comment]
不可以通过Activity哟！

**Comment by ManyullynSword** [comment]
可以通过爱玩机工具箱启动，就是有点繁琐

**Comment by ManyullynSword** [comment]
LSPosed的Activity名称为org.lsposed.manager.ui.activity.MainActivity

**Comment by ManyullynSword** [comment]
默认包名为com.android.shell

**Comment by ManyullynSword** [comment]
可我无法通过Activity唤起，Shell也不是能正常打开的应用

**Reply by KJZH001** [reply]
我明天尝试一下吧，或许也需要考虑用root强行拉起活动
不过我想应该是有办法唤起的，桌面打开应用程序也同样需要类似的事情，或许lsp需要有特殊的参数要附带上去（当然我们目前还不知道是什么）

**Comment by KJZH001** [comment]
Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=com.android.shell/.org.lsposed.manager.ui.activity.MainActivity }
Error type 3
Error: Activity class {com.android.shell/com.android.shell.org.lsposed.manager.ui.activity.MainActivity} does not exist.
redfin:/ #

看起来貌似是活动名称不对？

**Reply by KJZH001** [reply]
通过adb重新确定了一下，活动名称是对的，但确实拉不起来(汗

**Reply by ManyullynSword** [reply]
或者说，你手动安装一个启动器······

**Reply by ManyullynSword** [reply]
桌面

**Reply by ManyullynSword** [reply, marked as answer]
其实最方便就是直接安装Manager.apk，什么事情也没有，就是“可能”会泄露。

**Reply by KJZH001** [reply]
直接安装Manager.apk似乎也是个办法，至于泄露，靠隐藏应用列表吧（
不过这玩意官方还在维护吗？

**Reply by KJZH001** [reply]
>其实最方便就是直接安装Manager.apk，什么事情也没有，就是“可能”会泄露。

我后面去网上随便找了一个Manager.apk的安装包（github这边我似乎找不到它），不过安装后确实能够直接通过应用列表打开它了

**Reply by ManyullynSword** [reply]
Ruru 会发现，但你为什么要用网上找的 manager?

**Reply by ManyullynSword** [reply]
根目录是有 manager 的，在 LSPosed 模块的安装目录

**Reply by KJZH001** [reply]
/data/adb/lspd/manager.apk

你是指这个路径对吗？用网上找的manager是因为我不知道原来已经自带了有安装包

那么现在既然知道了自带的安装包那我就可以把网上找的换掉了

**Comment by tianjyan** [comment]
启动WSA后，试试这个
```bash
adb shell am start -a android.intent.action.MAIN -c 34c5cf58-a883-4a2d-b7eb-3467dcf632c1 -c org.lsposed.manager.LAUNCH_MANAGER -n com.android.shell/.BugreportWarningActivity com.android.shell
```

**Reply by ManyullynSword** [reply]
怎样把这个修改为WSA://的链接？

**Reply by tianjyan** [reply]
没找到任何文档描述wsa，如果有的话可以分享给我看下

**Comment by savfile** [comment]
> 启动WSA后，试试这个
> 
> ```shell
> adb shell am start -a android.intent.action.MAIN -c 34c5cf58-a883-4a2d-b7eb-3467dcf632c1 -c org.lsposed.manager.LAUNCH_MANAGER -n com.android.shell/.BugreportWarningActivity com.android.shell
> ```
简化版
```shell
adb shell am start -a android.intent.action.MAIN -c org.lsposed.manager.LAUNCH_MANAGER -n com.android.shell/.BugreportWarningActivity
```

**Comment by KJZH001** [comment]
感谢以上二位提供的adb指令

经过测试之后确实都能够顺利拉起lsposed管理器
